lib/rubocop/cop/mixin/configurable_naming.rb

```diff
@@ -22,9 +22,20 @@ def check_name(node, name, name_range)
         end
       end
 
-      def valid_name?(name)
+      def valid_name?(node, name)
         pattern = (style == :snake_case ? SNAKE_CASE : CAMEL_CASE)
-        name.match(pattern)
+        name.match(pattern) || class_emitter_method?(node, name)
+      end
+
+      # A class emitter method is a singleton method in a class/module, where
+      # the method has the same name as a class defined in the class/module.
+      def class_emitter_method?(node, name)
+        return false unless node.defs_type?
+        return false unless node.parent
```

Comment:
A small optimization would be to check if the name of the method is capitalized.